tests/cephfs/cephfs_vol_management/cephfs_vol_mgmt_subvolgroup_scenarios.py

```diff
@@ -0,0 +1,273 @@
+import random
+import string
+import traceback
+
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CephFS Subvolumegroup Test
+
+    This test suite validates the functionality of the `ceph fs subvolumegroup` command,
+    covering various scenarios including:
+
+    1. Creating a subvolumegroup with invalid and valid pool layout flags
+    2. Creating a subvolumegroup with desired and invalid modes
+    3. Creating a subvolumegroup with desired and invalid UID/GID
+    4. Creating and resizing a subvolumegroup with specific size
+    5. Creating a subvolumegroup with various options (mode, uid, gid, size)
+
+    """
+    try:
+        tc = "CEPH-83604079"
+        log.info(f"Running CephFS tests for - {tc}")
+        fs_util = FsUtils(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        config = kw.get("config")
+        fs_util.auth_list(clients)
+        build = config.get("build", config.get("rhbuild"))
+        fs_util.prepare_clients(clients, build)
+        client1 = clients[0]
+        vol_name = f"cephfs_{random.randint(0, 1000)}"
+        fs_util.create_fs(client1, vol_name)
+        mount_dir = "".join(
+            random.choice(string.ascii_lowercase + string.digits)
+            for _ in list(range(5))
+        )
+        fuse_mounting_dir = f"/mnt/cephfs_fuse_{mount_dir}/"
+        fs_util.fuse_mount(
+            [client1], fuse_mounting_dir, extra_params=f" --client_fs {vol_name}"
+        )
+        random_string = "".join(random.choices("abcdefghijklmnopqrstuvwxyz", k=5))
+        # Create a subvolumegroup with invalid pool layout
+        group_name = f"invalid_layout_group_{random_string}"
+        command = f"ceph fs subvolumegroup create {vol_name} {group_name} --pool_layout invalid"
```

Comment:
There is a new command to validate the creation using - " ceph fs subvolumegroup exist" - Can you use this to validate if the group is created ?

Reply:
@hkadam134 but I thought that `subvolumegroup exist {volume_name}` only checks if any subvolumegroup exists under the file system. if it is the case, I will validate that with `ceph fs subvolumegroup ls`

Reply:
ceph fs subvolumegroup exist {volume_name} is used for group delete, can you also include the same to validate after creation ? 
rest looks good.

Reply:
since ceph fs subvolumegroup eixst only tells if any subvolumegroup exist or not, how can we tell if the subvolumegroup is created..?

Reply:
Please have the below commands covered after creating and removing the subvolumegroup : 

[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup ls cephfs1
[]
[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup create cephfs1 gp1
[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup ls cephfs1
[
    {
        "name": "gp1"
    }
]
[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup exist cephfs1
subvolumegroup exists
[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup rm cephfs1 gp1
[root@ceph-hk-auto-4tjyb5-node11 ~]# ceph fs subvolumegroup exist cephfs1
no subvolumegroup exists
[root@ceph-hk-auto-4tjyb5-node11 ~]#

Reply:
I meant that along with subvolumegroup ls validation, "exist" should also be covered so that we exercise the cli in our tests.

Reply:
@hkadam134 I got what you mean. let me add that as well